lib/dashdash.d.ts

```diff
@@ -0,0 +1,261 @@
+     */
+    helpWrap?: boolean;
+
+    /**
+     * A default value used for this option, if the option isn't specified in argv.
+     */
+    default?: string;
+
+    /**
+     * If true, help output will not include this option.
+     */
+    hidden?: boolean;
+}
+
+export interface Group {
+    group: string;
+}
+
+export interface OptionType {
+    name: string;
+    takesArg: boolean;
+    helpArg: string;
+    parseArg(option: Option, optstr: string, arg: string): any;
+    array?: boolean;
+    arrayFlatten?: boolean;
+    default?: any;
+    completionType?: any;
+}
+
+export interface BashCompletionConfiguration {
+    /**
+     * The tool name.
+     */
+    name: string;
+
+    /**
+     * The array of dashdash option specs.
+     */
+    options?: Array<Option | Group>;
+
+    /**
+     * Extra Bash code content to add
+     * to the end of the "spec". Typically this is used to append Bash
+     * "complete_TYPE" functions for custom option types.
+     */
+    specExtra?: string;
+
+    /**
+     * Array of completion types for positional args (i.e. non-options).
+     * If not given, positional args will use Bash's 'default' completion.
+     */
+    argtypes?: string[];
+}
+
+export interface BashCompletionSpecConfiguration {
+    /**
+     * The array of dashdash option specs.
+     */
+    options: Array<Option | Group>;
+
+    /**
+     * A context string for the "local cmd*"
+     * vars in the spec. By default it is the empty string. When used to
+     * scope for completion on a *sub-command*.
+     */
+    context?: string;
+
+    /**
+     * By default
+     * hidden options and subcmds are "excluded". Here excluded means they
+     * won't be offered as a completion, but if used, their argument type
+     * will be completed. "Hidden" options and subcmds are ones with the
+     * `hidden: true` attribute to exclude them from default help output.
+     */
+    includeHidden?: boolean;
+
+    /**
+     * Array of completion types for positional args (i.e. non-options).
+     * If not given, positional args will use Bash's 'default' completion.
+     */
+    argtypes?: string[];
+}
+
+export interface HelpConfiguration {
+    /**
+     * Set to a number (for that many spaces) or a string for the literal indent.
+     * Default: 4
+     */
+    indent?: number | string;
+
+    /**
+     * Set to a number (for that many spaces) or a string for the literal indent.
+     * This indent applies to group heading lines, between normal option lines.
+     * Default: half length of `indent`
+     */
+    headingIndent?: number | string;
+
+    /**
+     * By default the names are sorted to put the short opts first (i.e. '-h, --help' preferred to '--help, -h').
+     * Set to 'none' to not do this sorting.
+     * Default: 'length'
+     */
+    nameSort?: string;
+
+    /**
+     * Note that reflow is just done on whitespace so a long token in the option help can overflow maxCol.
+     * Default: 80
+     */
+    maxCo?: number;

```

Comment:
Good news is I know the stupid mistake that caused this and it will be isolated to that interface.  Bad new is that it was a stupid mistake in the RegEx find/replace that I used to add the question mark that I didn't catch; I am glad you did.